src/main/scala/com/campudus/tableaux/database/model/TableauxModel.scala

```diff
@@ -32,96 +28,83 @@ class TableauxModel(override protected[this] val connection: DatabaseConnection)
 
   import TableauxModel._
 
-  val tableStruc = new TableStructure(connection)
-  val columnStruc = new ColumnStructure(connection)
-  val cellStruc = new CellStructure(connection)
-  val rowStruc = new RowStructure(connection)
-
-  val attachmentModel = AttachmentModel(connection)
-
-  def createTable(name: String): Future[Table] = for {
-    id <- tableStruc.create(name)
-  } yield Table(id, name)
-
-  def createCompleteTable(name: String, columnsNameAndType: Seq[CreateColumn], rowsValues: Seq[Seq[_]]): Future[CompleteTable] = for {
-    table <- createTable(name)
-    columnIds <- addColumns(table.id, columnsNameAndType) map { colSeq => colSeq.columns map { col => col.id } }
-    rowsWithColumnIdAndValue <- Future.successful {
-      if (rowsValues.isEmpty) {
-        Seq()
-      } else {
-        rowsValues map {
-          columnIds.zip(_)
-        }
-      }
-    }
-    _ <- addFullRows(table.id, rowsWithColumnIdAndValue)
-    completeTable <- getCompleteTable(table.id)
-  } yield completeTable
-
-  def deleteTable(tableId: TableId): Future[EmptyObject] = for {
-    _ <- tableStruc.delete(tableId)
-  } yield EmptyObject()
-
-  def deleteRow(tableId: TableId, rowId: RowId): Future[EmptyObject] = for {
-    _ <- rowStruc.delete(tableId, rowId)
-  } yield EmptyObject()
+  val cellStruc = new CellModel(connection)
+  val rowStruc = new RowModel(connection)
 
-  def addColumns(tableId: TableId, columns: Seq[CreateColumn]): Future[ColumnSeq] = for {
-    cols <- serialiseFutures(columns) {
-      case CreateSimpleColumn(name, kind, ordering) =>
-        addValueColumn(tableId, name, kind, ordering)
+  lazy val attachmentModel = AttachmentModel(connection)
+  private lazy val structureModel = StructureModel(connection)
```

Comment:
Why are the modifiers for each `*Model` different? (Two `lazy`, one `private`...)